### DISCUSSION TITLE
Is there a way to use UpdateRefitsEvery in evaluate?

### DISCUSSION THREAD

**Discussion by amurthy-sunysb** [alan-turing-institute/sktime#6851] [Q&A] [answered]
My understanding is that the backtesting enabled by  the `evaluate` function calls `fit `or `update ` based on the `strategy`.

I would like backtest a hybrid strategy that keeps updating the model for a few iterations and refits the model at a prescribed frequency.  This sounds like a job for `UpdateRefitsEvery`.  Can we implement this using the existing `evaluate ` function? -Thx!

**Comment by fkiraly** [comment, marked as answer]
Of course!

You can do this as follows:

* in `evaluate`, use `strategy="update"` and the expanding window splitter or a similar one.
* wrap the forecasters that you want to update less frequently in `UpdateRefitsEvery`.

This will call `update` every time the window progresses on the forecaster, but it will trigger a "true" `update` only at the lower frequency.

The choice to attach this to the model rather than to the evaluation strategy is because updating less frequently is primarily a modelling choice, rather than an evaluation choice. Interesting question: would you agree with this or not?

**Reply by amurthy-sunysb** [reply]
Thank you @fkiraly, this works!

In resource-constrained or cost-sensitive settings, we may want to evaluate how long can we put off retraining the model.

**Reply by fkiraly** [reply]
Yes, I know! That's why I wrote `UpdateRefitsEvery`, and it has served me well 😁
